modules/ROOT/pages/client-id-based-policies.adoc

```diff
@@ -14,48 +14,73 @@ endif::[]
 |500 - Bad response from authorization server, or WSDL SOAP Fault error
 |===
 
-The Client ID Enforcement Policy restricts access to a protected resource, by only allowing HTTP requests if credentials of client applications already xref:api-contracts-landing-page.adoc[registered to (contracted with) the API being targeted] are provided in each request.
+The Client ID Enforcement Policy restricts access to a protected resource, by only allowing HTTP requests if credentials of client applications already xref:api-contracts-landing-page.adoc[registered to (contracted with) the API being targeted] are provided in each request. In Anypoint Platform, the client ID and client secret pair is also known as credentials.
 
-The policy validates the Client ID and, optionally, the Client Secret, of a client application created within an Anypoint Platform Organization.
+After the Client ID Enforcement policy is applied, the Mule runtime engine (Mule) caches the information locally and updates it periodically. In addition to performance improvements, caching enables the policy to work offline during connectivity outages. 
 
-IMPORTANT: This policy will not validate successfully any client application credentials generated outside of the Anypoint Platform context.
+Regardless of the state of the management plane, the credentials are enforced with the existing data. Once the connectivity is re-established, the cache resumes periodical updates.
 
-NOTE: The set of Client ID and Client Secret are also known as credentials.
+//What information is stored or retrieved. -- Eva to send
+//Latest version has encryption. So the information is encrypted.
 
-= Analytics and  access tracking
-If you need to track client application access to your API, apply a policy that directly or indirectly makes sure that client ID is provided in the HTTP request (ie Client ID enforcement type policy).
+// How does the policy work?
+//Where can you get the inforamtion for the policy--Mariano to send me the information
 
-= Indirect client ID enforcement
-There are other provided policies that enforce client application credentials. For example:
 
-* xref:rate-limiting-and-throttling-sla-based-policies.adoc[Rate Limiting - SLA-Based Policies]
+Default policies that internally enforce client application credentials include:
```

Comment:
There are other default policies that internally enforce requests coming from registered client applications:

Reply:
Which? We should name them all.

Reply:
JWT was the one to add, I added in the comment and I can see you added it in the list.

Reply:
Awesome! Done!